shared/reducers/settings.tsx

```diff
@@ -169,6 +169,9 @@ const passwordActions: Container.ActionHandler<Actions, Types.State> = {
     password.error = undefined
     password.newPasswordConfirm = action.payload.password
   },
+  [SettingsGen.resetCheckPasswordIsCorrect]: draftState => {
+    draftState.checkPasswordIsCorrect = undefined
```

Comment:
Added this to reset the "checked password" state, which isn't reset to `undefined`